source/docs/software/basic-programming/cpp-units.rst

```diff
@@ -1,79 +1,116 @@
 
 .. code-block:: c++
 
-   auto area = 36_sq_m;
-   units::meter_t sideLength = units::math::sqrt(area);
+   units::meter_t distance = 6_ft;
 
-Performing Unit Operations
-^^^^^^^^^^^^^^^^^^^^^^^^^^
-The units library can also be used to perform operations between units, such as addition and multiplication.
+In short, we can use *any* unit of length in place of *any other* unit of length, anywhere in our code; the units library will automatically perform the correct conversion for us.
+
+Performing Arithmetic with Units
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Container types support all of the ordinary arithmetic operations of their underlying data type, with the added condition that the operation must be *dimensionally* sound.  Thus, addition must always be performed on two compatible container types:
 
 .. code-block:: c++
 
    // Add two meter_t values together
-   auto one = 5_m;
-   auto two = 7_m;
-   units::meter_t sum = one + two; // sum is 12_m
+   auto sum = 5_m + 7_m; // sum is 12_m
+
+   // Adds meters to feet; both are length, so this is fine
+   auto sum = 5_m + 7_ft;
+
+   // Tries to add a meter_t to a second_t, will throw a compile-time error
+   auto sum = 5_m + 7_s;
+
+Multiplication may be performed on any pair of container types, and yields the container type of a compound unit:
+
+.. note:: When a calculation yields a compound unit type, this type will only be checked for validity at the point of operation if the result type is specified explicitly.  If ``auto`` is used, this check will not occur.  For example, when we divide distance by time, we may want to ensure the result is, indeed, a velocity (i.e. ``units::meter_per_second_t``). If the return type is declared as ``auto``, this check will not be made.
 
 .. code-block:: c++
 
-   // Multiply two meter_t values to create a square_meter_t
-   auto one = 5_m;
-   auto two = 7_m;
-   units::square_meter_t product = one * two; // product is 35_sq_m
+   // Multiply two meter_t values, result is square_meter_t
+   auto product = 5_m * 7_m; // product is 35_sq_m
 
 .. code-block:: c++
 
-   // Divide a meter_t value by a second_t value to create a speed.
-   auto distance = 6_m;
-   auto time = 0.5_s;
+   // Divide a meter_t value by a second_t, result is a meter_per_second_t
+   units::meter_per_second_t speed = 6_m / 0.5_s; // speed is 12_mps
+
+``<cmath>`` Functions
+^^^^^^^^^^^^^^^^^^^^^
+
+Some ``std`` functions (such as ``clamp``) are templated to accept any type on which the arithmetic operations can be performed.  Quantities stored as container types will work with these functions without issue.
 
-   units::meter_per_second_t speed = distance / time; // speed is 12_mps
+However, other ``std`` functions work only on ordinary numerical types (e.g. ``double``).  The units library's ``units::math`` namespace contains wrappers for several of these functions that accept units. Examples of such functions include ``sqrt``, ``pow``, etc.
```

Comment:
Do you think it's worth mentioning that `units::math::pow` is templated on the power, as opposed to the std implementation which is not?

Reply:
Eh, not really.  If users are using that, they're probably deep enough to want to read the actual documentation.

Reply:
Fair enough.